packages/fxa-auth-server/lib/oauth/db/mysql/index.js

```diff
@@ -978,34 +978,37 @@ MysqlStore.prototype = {
         if (conn._fxa_initialized) {
           return resolve(conn);
         }
+
         // Enforce sane defaults on every new connection.
         // These *should* be set by the database by default, but it's nice
         // to have an additional layer of protection here.
-        conn.query('SELECT @@sql_mode AS mode', function(err, rows) {
-          if (err) {
-            return reject(err);
-          }
-          var modes = rows[0]['mode'].split(',');
-          var needToSetMode = false;
-          REQUIRED_SQL_MODES.forEach(function(requiredMode) {
-            if (modes.indexOf(requiredMode) === -1) {
-              modes.push(requiredMode);
-              needToSetMode = true;
+        const query = P.promisify(conn.query, { context: conn });
+        return resolve(
+          (async () => {
+            // Always communicate timestamps in UTC.
+            await query("SET time_zone = '+00:00'");
+            // Always use full 4-byte UTF-8 for communicating unicode.
+            await query('SET NAMES utf8mb4 COLLATE utf8mb4_unicode_ci;');
```

Comment:
The oauth-server was previously not doing an explicit `SET NAMES` during connection setup, but both auth-server and profile-server were. I've added it here for consistency.

Well...I say "consistency" but it actually uses a different collation order than the other two, for what I assume are really fascinating historical reasons. See `REQUIRED_CHARSET` at the top of the file.